include/miroil/miroil/mirbuffer.h

```diff
@@ -21,41 +21,38 @@
 #include <mir/version.h>
 #include <memory>
 
-#include <GL/gl.h>
-
-namespace mir { namespace graphics { class Buffer; }}
+namespace mir {
+    namespace graphics {
+        class Buffer;
+
+        namespace gl {
+            class Texture;
+        }
+    }
+}
 
 namespace miroil
 {
 class GLBuffer
 {
 public:
-    GLBuffer();
     ~GLBuffer();
     explicit GLBuffer(std::shared_ptr<mir::graphics::Buffer> const& buffer);
 
-    operator bool() const;
     bool has_alpha_channel() const;
     mir::geometry::Size size() const;
+    void bind();
 
-    void reset();
     void reset(std::shared_ptr<mir::graphics::Buffer> const& buffer);
-    void bind();
-#if MIR_SERVER_VERSION < MIR_VERSION_NUMBER(2, 3, 0)
-    void gl_bind_tex();
-#endif
 
-private:
-    void init();
-    void destroy();
+    void reset();
+    bool empty();
+
+    static std::shared_ptr<GLBuffer> from_mir_buffer(std::shared_ptr<mir::graphics::Buffer> const& buffer);
 
+protected:
```

Comment:
Why not `private`? There are no virtual methods and CopyAssign allows slicing - this looks like there's no expectation of derived classes

Reply:
was left from qtmir, will change it

Reply:
done